airflow/providers/google/cloud/log/gcs_task_handler.py

```diff
@@ -132,7 +134,10 @@ def close(self):
             # read log and remove old logs to get just the latest additions
             with open(local_loc) as logfile:
                 log = logfile.read()
-            self.gcs_write(log, remote_loc)
+            success = self.gcs_write(log, remote_loc)
+            keep_local = conf.getboolean('logging', 'KEEP_LOCAL_LOGS')
+            if success and not keep_local:
+                shutil.rmtree(os.path.dirname(local_loc))
```

Comment:
You're implementing the same cleanup recipe several times on the back of a global config, both of which are indicators that this is a good candidate for logic that should live in a super class. Doing it ad hoc like this leaves us open for future developers of remote logging classes to forget or mis-implement this logic. The individual remote logging classes should only be responsible for doing the upload to their respective service, they shouldn't have to re-implement the cleanup.

It is possible to teach `FileTaskHandler` to do this, but it would be tricky to make it work in both cases and is a bit smelly. It's maybe time for a new super class `RemoteTaskHandler`?

Reply:
I avoided putting it on `FileTaskHandler` because it was a bit messy but having a `RemoteFileTaskHandler` could indeed help there.